File: modules/aidj/aidj相关/aidjmix-autoplaylist-python/aidjmix/transitions.py
from typing import List
from .types import TrackFeature, TransitionPlan, PlaylistItem
from .config import TRANSITION

def plan_transitions(seq: List[TrackFeature], techno: bool=True) -> TransitionPlan:
    items: List[PlaylistItem] = []
    for i, cur in enumerate(seq):
        nxt = seq[i+1] if i+1 < len(seq) else None
        cross = TRANSITION["techno_crossfade_beats"] if (techno and nxt) else (TRANSITION["default_crossfade_beats"] if nxt else 0)
        start = cur.cueInSec if cur.cueInSec is not None else 0.0
        end = cur.cueOutSec if cur.cueOutSec is not None else cur.durationSec

        item = PlaylistItem(
            track=cur,
            startAt=start,
            endAt=end,
            crossfadeBeats=cross,
            automation=[]
        )
        if nxt:
            item.automation.append({"type":"filter","mode":"hipass_ramp","from_hz":80,"to_hz":260,"duration_beats":cross})
        items.append(item)
    total = sum((it.endAt or it.track.durationSec)-(it.startAt or 0) for it in items)
    return TransitionPlan(items=items, totalSec=total, avgScore=0.0)

File: modules/aidj/aidj相关/aidjmix-autoplaylist-api-ts/src/transitions.ts
import { TrackFeature, TransitionPlan, PlaylistItem } from './types.js'; import { TRANSITION } from './config.js';
export function planTransitions(seq:TrackFeature[], techno=true):TransitionPlan{
  const items:PlaylistItem[]=[]; for(let i=0;i<seq.length;i++){ const cur=seq[i], nxt=seq[i+1];
    const cross=nxt? (techno? TRANSITION.technoCrossfadeBeats: TRANSITION.defaultCrossfadeBeats): 0;
    const it:PlaylistItem={ track:cur, startAt: cur.cueInSec ?? 0, endAt: cur.cueOutSec ?? cur.durationSec, crossfadeBeats: cross,
      automation: nxt? [{type:'filter',mode:'hipass_ramp',from_hz:80,to_hz:260,duration_beats:cross}]: [] };
    items.push(it); }
  const total=items.reduce((s,it)=>s+((it.endAt??it.track.durationSec)-(it.startAt??0)),0); return { items, totalSec: total, avgScore: 0 }; }
